llarp/router/router.cpp

```diff
@@ -770,6 +770,47 @@ namespace llarp
 
     _linkManager.CheckPersistingSessions(now);
 
+    if (not IsServiceNode())
+    {
+      const auto gateway = GetDefaultGateway();
+      if (m_CurrentGateway != gateway)
+      {
+        // changed gateways
+        if (m_CurrentGateway.empty())
+        {
+          LogInfo("found default gateway: ", gateway);
+        }
+        else if (not gateway.empty())
+        {
+          LogInfo("default gateway changed from ", m_CurrentGateway, " to ", gateway);
+        }
+        else
+        {
+          LogError("Network is down");
+        }
+        // unpoke current routes
+        std::unordered_set<std::string> holes;
+
+        for (const auto& [ip, gw] : m_PokedRoutes)
+        {
+          // save hole
+          holes.emplace(ip);
+          // unpoke route
+          net::DelRoute(ip, gw);
+        }
+        m_PokedRoutes.clear();
+
+        if (not gateway.empty())
```

Comment:
Seems this is basically "our gateway changed, so check if that means we've just lost internet connectivity entirely".  In that case, will the routes we removed above be recreated properly when connectivity is restored?  Or will new first hops be selected rendering that moot?

Reply:
yes they will be recreated later on demand.